core/src/main/java/org/apache/accumulo/core/metadata/StoredTabletFile.java

```diff
@@ -71,4 +92,40 @@ public void validate(String reference) {
           + " does not match what was in the metadata: " + metadataEntry);
     }
   }
+
+  @Override
+  public int compareTo(StoredTabletFile o) {
+    if (equals(o)) {
+      return 0;
+    } else {
+      return metadataEntry.compareTo(o.metadataEntry);
+    }
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    StoredTabletFile that = (StoredTabletFile) o;
+    return Objects.equals(metadataEntry, that.metadataEntry);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(metadataEntry);
+  }
+
+  @Override
+  public String toString() {
+    return metadataEntry;
```

Comment:
Wondering if TabletFile.toString() and this would ever be equivalent. If they could be, should we add something to the toString() output that makes them different in case there is code that is comparing the toString output for equivalence instead of using equals()?

Reply:
Same comment goes for UnreferencedTabletFile.toString

Reply:
Anything you would add would probably break things that are reading the `toString()`, so you'd be potentially creating more runtime bugs to fix one potential runtime bug. Not sure if that's worth it. Either way, would need to track down who is calling toString on these, and figuring out what they're using it for in each case.

Reply:
I searched and I didn't see any places where there is a comparison done, it's usually used for logging and exceptions. However, there is actually one spot in FileCompactor where it's [used](https://github.com/apache/accumulo/blob/0017fd8949cc1bc7253b8e90313d4fd1b71e2241/server/base/src/main/java/org/apache/accumulo/server/compaction/FileCompactor.java#L195) as the actual file so changing this would break things. That spot should actually be fixed so I will update this PR to use getPathStr() instead.